### PR TITLE
feat: playbook AI Action

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/mission-control/issues/1642

### CODE REVIEW

File: fixtures/playbooks/ai-diagnose-kubernetes-resource.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      ai:
+        backend: openai
+        model: gpt-4o
+        apiKey:
+          valueFrom:
+            secretKeyRef:
+              name: openai
+              key: API_KEY
+        prompt: "{{.params.prompt}}"
+        config: "{{.config.id}}"
```
Comment (moshloop) [inline fixtures/playbooks/ai-diagnose-kubernetes-resource.yaml:27]: Why do we need this ? The config is added as a parameter ?

File: playbook/actions/ai.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	return &AIActionResult{Logs: response}, nil
+}
+
+func buildPrompt(ctx context.Context, prompt string, spec v1.AIActionContext) (string, error) {
+	pctx, err := getPromptContext(ctx, spec)
+	if err != nil {
+		return "", fmt.Errorf("failed to get prompt context: %w", err)
+	}
+
+	paragraphs := []string{
```
Comment (moshloop) [inline playbook/actions/ai.go:62]: This should be formatted into JSON:

```yaml
models.ConfigItem
config: json.RawMessage
changes: []models.ConfigChange
analysis: []models.ConfigInsight
graph: []struct{
{from, to, type, name}
},
```

Comment (moshloop) [reply playbook/actions/ai.go:62]:
```
	[]llms.MessageContent{
			{
				Role: llms.ChatMessageTypeHuman,
				Parts: []llms.ContentPart{

					llms.BinaryPart("application/json", json),
					llms.TextPart("Please tell me the string on the box."),
				},
			},
		},
```

Comment (adityathebe) [reply playbook/actions/ai.go:62]: why do we want to send this as json? 
what we currently have is more concise and shows the relationship explicitly.

Comment (adityathebe) [reply playbook/actions/ai.go:62]:
```go
llms.BinaryPart("application/json", json)
```

this is not allowed

File: config/schemas/openapi.go
Comment (adityathebe) [inline config/schemas/openapi.go:1]: Bringing this in from duty.
It's a hassle to update schemas as we need to first make a duty release with an unmerged schema.

File: playbook/actions/ai.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+	"github.com/flanksource/incident-commander/api"
+	v1 "github.com/flanksource/incident-commander/api/v1"
+	"github.com/flanksource/incident-commander/llm"
+)
+
+type AIAction struct{}
+
+type AIActionResult struct {
+	Logs string `json:"logs,omitempty"` // TODO: only naming this "logs" because the frontend has proper formatted display for this field
```
Comment (moshloop) [inline playbook/actions/ai.go:24]: I think we need to return a markdown response, 

We could probably convert  https://codesandbox.io/p/sandbox/markdown-to-slack-converter-iphgn?file=%2Fsrc%2Findex.js%3A25%2C56 into a go function and then just call `markdownToSlackBlocks`

**Pull request merged**